lib/Sema/CSGen.cpp

```diff
@@ -1345,8 +1345,10 @@ namespace {
       // We may, alternatively, want to use a type variable in that case,
       // and possibly infer the type of the variable that way.
       CS.getTypeChecker().validateDecl(E->getDecl());
-      if (E->getDecl()->isInvalid())
+      if (E->getDecl()->isInvalid()) {
+        CS.setType(E, E->getDecl()->getInterfaceType());
```

Comment:
The fixme notes spawning and solving a type variable here would potentially provide better diagnostics instead of no diagnostics.